lib/logstash/filters/mutate.rb

```diff
@@ -233,6 +245,18 @@ def remove(event)
   end # def remove
 
   private
+  def select(event)
+    replacement = LogStash::Event.new
+
+    @select.each do |field|
+      field = event.sprintf(field)
+      next unless event.include?(field)
+      replacement[field] = event[field]
+    end
+    event.overwrite replacement
```

Comment:
Not completely sure about the best way, but I would say this needs some work on the logstash core, to prevent override to touch this variables. Then your code is going to work as expected without more changes.

I would say, changing ```@timestamp``` as it happens now, is not acceptable as it might screw up some stuff.

@talevy what do you think?